src/main/java/org/eclipse/cargotracker/interfaces/booking/sse/RealtimeCargoTrackingService.java

```diff
@@ -53,11 +57,14 @@ public void close() {
   }
 
   public void onCargoUpdated(@ObservesAsync @CargoUpdated Cargo cargo) {
-    logger.log(Level.FINEST, "SSE event broadcast for cargo: {0}", cargo);
-    broadcaster.broadcast(cargoToSseEvent(cargo));
+    if (broadcaster != null && (sse.newEventBuilder() != null)) {
```

Comment:
Is the SSE null check necessary or redundant?

Reply:
The null broadcaster check seems to be redundant but the sseEventBuilder check is necessary for OL.